test/parallel/test-runner-exit-code.js

```diff
@@ -10,11 +11,17 @@ if (process.argv[2] === 'child') {
     test('passing test', () => {
       assert.strictEqual(true, true);
     });
-  } else {
+  } else if (process.argv[3] === 'fail') {
     assert.strictEqual(process.argv[3], 'fail');
     test('failing test', () => {
       assert.strictEqual(true, false);
     });
+  } else if (process.argv[3] === 'never_ends') {
+    assert.strictEqual(process.argv[3], 'never_ends');
+    test('never ending test', () => {
+      return setTimeout(100_000_000);
+    });
+    process.kill(process.pid, 'SIGINT');
   }
```

Comment:
nit: this could help future iteration of this test e.g. if someone makes a typo, it's helpful to see the test crash rather than silently pass
```suggestion
  } else assert.fail('unreachable');
```